sql/2020/06_Third_Parties/percent_of_third_party_loaded_before_DOMContentLoaded.sql

```diff
@@ -0,0 +1,62 @@
+#standardSQL
+# Percent of third-party requests loaded before DOM Content Loaded event
+
+CREATE TEMP FUNCTION get_load_end_time(payload STRING)
+RETURNS INT64 LANGUAGE js AS '''
+  try {
+    var $ = JSON.parse(payload);
+    return $._load_end
+  } catch (e) {
+    return false;
+  }
+''';
+
+WITH requests AS (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    page,
+    url,
+    get_load_end_time(payload) as load_end
```

Comment:
You could extract the value in SQL and avoid the need for the JS function (not tested)
```suggestion
    SAFE_CAST(JSON_EXTRACT_SCALAR(payload, '$._load_end') AS INT64) AS load_end
```